extmod/modubinascii.c

```diff
@@ -89,14 +88,89 @@ mp_obj_t mod_binascii_unhexlify(mp_obj_t data) {
 }
 MP_DEFINE_CONST_FUN_OBJ_1(mod_binascii_unhexlify_obj, mod_binascii_unhexlify);
 
+static const char *base64codes = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
```

Comment:
> There's a reason everyone uses the static array

Yes, it was given above - cut&paste programming ;-).

> the base64 text codes aren't contiguous on the ASCII table

Yes, there're 3 contiguous subranges, approaching rocket science ;-)

> every simple lookup changes to at least one "if" statement, possibly 2 or 3.

3 sub-ranges are 6 comparisons. Let's round up to 10 for other things. Your strchr() does worst-case 64 comparisons! So, not only direct code will likely have smaller size, it will certainly will be more perfomant.

> I'll give it a shot and check the results.

Thanks. And feel free to argue for example that your impl is already good enough and optimizations can be done later. Counter-argument will be that usually it never comes to real optimizations when only deoptimized code is committed. That's why we do it differently in uPy. So, please rather take that as encouragement to challenge a way "everyone does it". It's lotsa fun. That's why uPy is so cool.